### ISSUE TITLE
Fix warnings with gcc 6.1

### ISSUE THREAD

**Issue by TheMarex** [Project-OSRM/osrm-backend#2596]
```
/home/patrick/Code/osrm/src/engine/guidance/lane_processing.cpp: In lambda function:
/home/patrick/Code/osrm/src/engine/guidance/lane_processing.cpp:67:19: warning: unused variable ‘previous_inst’ [-Wunused-variable]
             auto &previous_inst = previous.maneuver.instruction;
                                ^~~~~~~~~~~~~
/home/patrick/Code/osrm/src/engine/guidance/post_processing.cpp: In function ‘std::vector<osrm::engine::guidance::RouteStep> osrm::engine::guidance::buildIntersections(std::vector<osrm::engine::guidance::RouteStep>)’:
/home/patrick/Code/osrm/src/engine/guidance/post_processing.cpp:1085:45: warning: comparisons like ‘X<=Y<=Z’ do not have their mathematical meaning [-Wparentheses]
                 BOOST_ASSERT(step_index > 0 && step_index < step_index + 1 < steps.size());
                                                   ~~~~~~~~~~~^~~~~~~~~~~~~~~~
```

/cc @MoKob

**Comment by MoKob** [comment]
@daniel-j-h the lane_processing was one of yours. Any intention to do something with it?

**Comment by TheMarex** [comment]
Fixed by #2597.

**Comment by daniel-j-h** [comment]
Yes, leftover from prototyping. Sorry for that.
